src/common/schedulers/slurm_commands.py

```diff
@@ -249,7 +293,16 @@ def get_nodes_info(nodes="", command_timeout=DEFAULT_GET_INFO_COMMAND_TIMEOUT):
     Retrieve SlurmNode list from slurm nodelist notation.
 
     Sample slurm nodelist notation: queue1-dy-c5_xlarge-[1-3],queue2-st-t2_micro-5.
+    If no nodes argument is provided, this function considers only nodes in partitions managed by ParallelCluster.
+    It is responsibility of the caller to pass a nodes argument with only nodes in partitions managed by
+    ParallelCluster.
+
+    TODO: we can consider building a filter to be used in case a nodes argument is passed, in order to exclude nodes
+     not managed by ParallelCluster.
     """
+    if nodes == "":
+        nodes = _get_all_partition_nodes(",".join(PartitionNodelistMapping.instance().get_partitions()))
```

Comment:
Do we still want this with the logic we chose to implement (i.e. _allowing_ users to add nodes in PC-managed partitions)?